RecoTracker/TkSeedGenerator/plugins/DeepCoreSeedGenerator.cc

```diff
@@ -0,0 +1,638 @@
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+
+#include <boost/range.hpp>
+#include <boost/foreach.hpp>
+#include "boost/multi_array.hpp"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "SimDataFormats/Vertex/interface/SimVertex.h"
+
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+
+#include "TTree.h"
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+//
+// class declaration
+//
+
+// If the analyzer does not use TFileService, please remove
+// the template argument to the base class so the class inherits
+// from  edm::one::EDAnalyzer<> and also remove the line from
+// constructor "usesResource("TFileService");"
+// This will improve performance in multithreaded jobs.
+
+DeepCoreSeedGenerator::DeepCoreSeedGenerator(const edm::ParameterSet& iConfig)
+    :
+
+      vertices_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("vertices"))),
+      pixelClusters_(
+          consumes<edmNew::DetSetVector<SiPixelCluster>>(iConfig.getParameter<edm::InputTag>("pixelClusters"))),
+      cores_(consumes<edm::View<reco::Candidate>>(iConfig.getParameter<edm::InputTag>("cores"))),
+      ptMin_(iConfig.getParameter<double>("ptMin")),
+      deltaR_(iConfig.getParameter<double>("deltaR")),
+      chargeFracMin_(iConfig.getParameter<double>("chargeFractionMin")),
+      centralMIPCharge_(iConfig.getParameter<double>("centralMIPCharge")),
+      pixelCPE_(iConfig.getParameter<std::string>("pixelCPE")),
+
+      weightfilename_(iConfig.getParameter<edm::FileInPath>("weightFile").fullPath()),
+      inputTensorName_(iConfig.getParameter<std::vector<std::string>>("inputTensorName")),
+      outputTensorName_(iConfig.getParameter<std::vector<std::string>>("outputTensorName")),
+      nThreads(iConfig.getParameter<unsigned int>("nThreads")),
+      singleThreadPool(iConfig.getParameter<std::string>("singleThreadPool")),
+      probThr(iConfig.getParameter<double>("probThr"))
+
+{
+  produces<TrajectorySeedCollection>();
+  produces<reco::TrackCollection>();
+}
+
+DeepCoreSeedGenerator::~DeepCoreSeedGenerator() {}
+
+void DeepCoreSeedGenerator::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  auto result = std::make_unique<TrajectorySeedCollection>();
+  auto resultTracks = std::make_unique<reco::TrackCollection>();
+
+  //-------------------TensorFlow setup - session (1/2)----------------------//
+  tensorflow::setLogging("3");
+  graph_ = tensorflow::loadGraphDef(weightfilename_);
+  tensorflow::SessionOptions sessionOptions;
+  tensorflow::setThreading(sessionOptions, nThreads, singleThreadPool);
+  session_ = tensorflow::createSession(graph_, sessionOptions);
```

Comment:
Just don't wrap the `tensorflow::GraphDef` to `std::atomic` but make it `mutable` instead (the use of `atomic` in `DeepMETProducer` is really circumventing `mutable`, and neither would be needed if our `tensorflow::createSession()` would take `GrapDef` as `const` as it really should).

(apologies to @cms-sw/reconstruction-l2 for sounding like a broken record)

Reply:
I'm sorry, because I'm not a TensorFlow expert I will follow the DeepMET step by step. @makortel the only change you're suggesting is define the graph as:
`mutable tensorflow::GraphDef* graph_def
`
Or something more?

Reply:
Yes, `mutable tensorflow::GraphDef* graph_def` until the interface of `tensorflow::createSession()` gets fixed.

Reply:
ok thank you very much. Implemented in this way and tested, everything seems working fine.